### PR TITLE
Int: Cleanup tests and dependencies of shared and runtime library

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
All the tests were left in SDK library.

# This PR's solution
Move proper tests to Shared and runtime library so it can be properly tested

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: projects/orquestra-sdk/tests/runtime/performance/test_cli_perf.py
Comment (alexjuda) [inline projects/orquestra-sdk/tests/runtime/performance/test_cli_perf.py:1]: Is there a reason why this file wasn't moved under `projects/orquestra-workflow-runtime/tests`?

File: projects/orquestra-sdk/tests/runtime/ray/conftest.py
Comment (alexjuda) [inline projects/orquestra-sdk/tests/runtime/ray/conftest.py:1]: Is there a reason why `projects/orquestra-sdk/tests/runtime/ray` directory wasn't moved under `projects/orquestra-workflow-runtime/tests`?

**Review by alexjuda — APPROVED** [review]
Found few potentially missed files

File: projects/orquestra-sdk/tests/runtime/performance/test_cli_perf.py
Comment (SebastianMorawiec) [reply projects/orquestra-sdk/tests/runtime/performance/test_cli_perf.py:1]: Yes - this test is testing CLI which is only part of SDK client side, its not available on runtime

File: projects/orquestra-sdk/tests/runtime/ray/conftest.py
Comment (SebastianMorawiec) [reply projects/orquestra-sdk/tests/runtime/ray/conftest.py:1]: As for anything left in runtime/ray - those tests are higher-level requiring us to use client-level concepts, like imports.  After we invent a way to generate IRs on flight inside RUNTIME package, we can move them all then, as now handling and maintanance would be very-very costly.

**Pull request merged**